### DISCUSSION TITLE
Duda con primer pago de 0€

### DISCUSSION THREAD

**Discussion by coriaweb** [creagia/laravel-redsys#67] [Q&A] [answered]
Hola!

Tengo una aplicación donde todos los pagos "la mayoría" serán tokenizados, actualmente lo tenemos así pero en un sistema wordpress, aquí en laravel ando dandome cabezazos jajaja

Os cuento:

1º Hay un apartado para añadir las tarjetas que quieres guardar para poder realizar los pagos en nuestro sistema. Esto debería de enviar una petición a redsys con importe de 0€, pero si lo envio a 0, obtengo este error:

`Creagia\Redsys\Support\RequestParameters::__construct(): Argument #1 ($amountInCents) not passed`

nada mas pongo por ejemplo 100, pasa bien, lo tengo asi en el modelo:

```
public function getTotalAmount(): int
    {
        return (int)(1 * 100); // Convertimos el total a céntimos
    }
    
    y asi el controlador:
    
    public function store(Request $request)
    {
        $user = Auth::user(); // Obtenemos el usuario logueado

        // Crear una solicitud de Redsys asociada con la factura
        $redsysRequest = $user->createRedsysRequest(
            productDescription: 'Autorización inicial',
            payMethod: PayMethod::Card, // Usamos el enum PayMethod::Card
        )->requestingCardToken(
            CofType::Recurring
        )->storeCardOnModel(
            $user
        );

        return $redsysRequest->redirect();
    }
```
    
    
¿Que es lo que estoy haciendo mal o como podemos hacer el primero de 0€ para que se quede en nuestro sistema el token y poder hacer cobros de las futuras suscripciones?
    
Saludos y gracias!

**Comment by dtorras** [comment, marked as answer]
Hola @coriaweb! Gracias por el mensaje. He encontrado un error en el package que no permitía enviar una request con importe zero.

Ya debería estar solucionado en las últimas release 
https://github.com/creagia/redsys-php/releases/tag/3.0.1
https://github.com/creagia/laravel-redsys/releases/tag/3.0.1

En tu aplicación, después de ejecutar `composer update` y comprobar que se ha actualizado a la versión 3.0.1, debería funcionar.

**Comment by coriaweb** [comment]
Ahora si que si!

https://i.gyazo.com/90177d01ee68038d84c7dffb7fe6c8a9.png

Gracias!!!

Debo decir, que me ha costado coger el hilo con la documentación y con el tipo de transacciones, el banco me comentó que tengo que enviar una excepción MIT y no se que mas.. jejeje. De momento en redirección para ir programando me sirve, pero.. ¿Tienes algo de documentación que ayude a que sea integrada dentro del sitio sin redireccionar? Jajajaja ¿Estoy complicandolo eh? kajajaja

**Comment by dtorras** [comment]
De entrada la documentación de Redsys es un poco complicada pero cuando te vas familiarizando con los conceptos es más fácil encontrar lo que se necesita.

Si te sirve, la excepción MIT es un tema que se ha comentado en este issue https://github.com/creagia/laravel-redsys/issues/41
